src/lib/swapper/swappers/ZrxSwapper/getZrxTradeQuote/getZrxTradeQuote.ts

```diff
@@ -1,138 +1,124 @@
+      takerAddress: receiveAddress,
+      affiliateAddress: AFFILIATE_ADDRESS, // Used for 0x analytics
+      skipValidation: true,
+      feeRecipient: getTreasuryAddressForReceiveAsset(buyAsset.assetId), // Where affiliate fees are sent
+      buyTokenPercentageFee: convertBasisPointsToDecimalPercentage(affiliateBps).toNumber(),
+    },
+  })
 
   if (maybeZrxPriceResponse.isErr()) return Err(maybeZrxPriceResponse.unwrapErr())
-
-  const data = maybeZrxPriceResponse.unwrap()
+  const { data } = maybeZrxPriceResponse.unwrap()
 
   const useSellAmount = !!sellAmount
   const rate = useSellAmount ? data.price : bn(1).div(data.price).toString()
 
-  // add gas limit buffer to account for the fact we perform all of our validation on the trade quote estimations
-  // which are inaccurate and not what we use for the tx to broadcast
-  const gasLimit = bnOrZero(data.gas).times(1.2)
+  // don't show buy amount if less than min sell amount
+  const isSellAmountBelowMinimum = bnOrZero(normalizedSellAmount).lt(minimumCryptoBaseUnit)
+  const buyAmountCryptoBaseUnit = isSellAmountBelowMinimum ? '0' : data.buyAmount
 
   // 0x approvals are cheaper than trades, but we don't have dynamic quote data for them.
   // Instead, we use a hardcoded gasLimit estimate in place of the estimatedGas in the 0x quote response.
-  const txFee = await (async () => {
-    if (optimism.isOptimismChainAdapter(adapter)) {
-      const { average, l1GasPrice } = await adapter.getGasFeeData()
-
-      // account for l1 transaction fees for optimism
-      const l1TxFee = bn(OPTIMISM_L1_SWAP_GAS_LIMIT).times(l1GasPrice)
-
-      return gasLimit.times(average.gasPrice).plus(l1TxFee)
-    }
-
+  try {
     const { average } = await adapter.getGasFeeData()
+    const networkFeeCryptoBaseUnit = calcNetworkFeeCryptoBaseUnit({
+      ...average,
+      eip1559Support,
+      // add gas limit buffer to account for the fact we perform all of our validation on the trade quote estimations
+      // which are inaccurate and not what we use for the tx to broadcast
+      gasLimit: bnOrZero(data.gas).times(1.2).toFixed(),
+      l1GasLimit: OPTIMISM_L1_SWAP_GAS_LIMIT,
+    })
 
-    const txFee = gasLimit.times(average.gasPrice)
-
-    return txFee
-  })()
-
-  const tradeQuote: TradeQuote<ZrxSupportedChainId> = {
-    minimumCryptoHuman: minimumAmountCryptoHuman,
-    steps: [
-      {
-        allowanceContract: data.allowanceTarget,
-        buyAsset,
-        sellAsset,
-        accountNumber,
-        rate,
-        feeData: {
-          networkFeeCryptoBaseUnit: txFee.toFixed(0),
-          protocolFees: {},
+    return Ok({
+      minimumCryptoHuman,
+      steps: [
+        {
+          allowanceContract: data.allowanceTarget,
+          buyAsset,
+          sellAsset,
+          accountNumber,
+          rate,
+          feeData: {
+            networkFeeCryptoBaseUnit,
+            protocolFees: {},
+          },
+          buyAmountBeforeFeesCryptoBaseUnit: buyAmountCryptoBaseUnit,
```

Comment:
Is this the same behavior as before? We were getting the `buyAmount` directly from `data` previously

Reply:
Not the same, we will use `data.buyAmount` if over the minimum amount, otherwise set it to `0` to imply you are unable to trade because it is less than min sell amount.

Before -> After example:
![image](https://github.com/shapeshift/web/assets/35275952/dd7b3742-d635-42e4-8dbe-63e44254274b)

Reply:
There would need to be additional data passed to the components to show `Insufficient sell amount` and also retain the buy amount returned from the quote, so standardizing current behavior with `0` buy amount when less than min for now